beacon-chain/rpc/eth/beacon/blocks.go

```diff
@@ -942,70 +943,63 @@ func (bs *Server) submitPhase0Block(ctx context.Context, phase0Blk *ethpbv1.Beac
 	if err != nil {
 		return status.Errorf(codes.InvalidArgument, "Could not convert block to v1 block")
 	}
-	wrappedAltairBlk, err := blocks.NewSignedBeaconBlock(v1alpha1Blk)
-	if err != nil {
-		return status.Errorf(codes.InvalidArgument, "Could not prepare block")
-	}
-
-	root, err := altairBlk.HashTreeRoot()
+	_, err = bs.V1Alpha1ValidatorServer.ProposeBeaconBlock(ctx, &eth.GenericSignedBeaconBlock{
+		Block: &eth.GenericSignedBeaconBlock_Altair{
+			Altair: v1alpha1Blk,
+		},
+	})
 	if err != nil {
-		return status.Errorf(codes.InvalidArgument, "Could not tree hash block: %v", err)
+		return status.Errorf(codes.Internal, "Could not propose block: %v", err)
 	}
-
-	return bs.submitBlock(ctx, root, wrappedAltairBlk)
+	return nil
 }
 
 func (bs *Server) submitBellatrixBlock(ctx context.Context, bellatrixBlk *ethpbv2.BeaconBlockBellatrix, sig []byte) error {
 	v1alpha1Blk, err := migration.BellatrixToV1Alpha1SignedBlock(&ethpbv2.SignedBeaconBlockBellatrix{Message: bellatrixBlk, Signature: sig})
 	if err != nil {
 		return status.Errorf(codes.InvalidArgument, "Could not convert block to v1 block")
 	}
-	wrappedBellatrixBlk, err := blocks.NewSignedBeaconBlock(v1alpha1Blk)
-	if err != nil {
-		return status.Errorf(codes.InvalidArgument, "Could not prepare block")
-	}
-
-	root, err := bellatrixBlk.HashTreeRoot()
+	_, err = bs.V1Alpha1ValidatorServer.ProposeBeaconBlock(ctx, &eth.GenericSignedBeaconBlock{
+		Block: &eth.GenericSignedBeaconBlock_Bellatrix{
+			Bellatrix: v1alpha1Blk,
+		},
+	})
 	if err != nil {
-		return status.Errorf(codes.InvalidArgument, "Could not tree hash block: %v", err)
+		return status.Errorf(codes.Internal, "Could not propose block: %v", err)
 	}
-
-	return bs.submitBlock(ctx, root, wrappedBellatrixBlk)
+	return nil
 }
 
 func (bs *Server) submitCapellaBlock(ctx context.Context, capellaBlk *ethpbv2.BeaconBlockCapella, sig []byte) error {
 	v1alpha1Blk, err := migration.CapellaToV1Alpha1SignedBlock(&ethpbv2.SignedBeaconBlockCapella{Message: capellaBlk, Signature: sig})
 	if err != nil {
 		return status.Errorf(codes.InvalidArgument, "Could not convert block to v1 block")
 	}
-	wrappedCapellaBlk, err := blocks.NewSignedBeaconBlock(v1alpha1Blk)
-	if err != nil {
-		return status.Errorf(codes.InvalidArgument, "Could not prepare block")
-	}
-
-	root, err := capellaBlk.HashTreeRoot()
+	_, err = bs.V1Alpha1ValidatorServer.ProposeBeaconBlock(ctx, &eth.GenericSignedBeaconBlock{
+		Block: &eth.GenericSignedBeaconBlock_Capella{
+			Capella: v1alpha1Blk,
+		},
+	})
 	if err != nil {
-		return status.Errorf(codes.InvalidArgument, "Could not tree hash block: %v", err)
+		return status.Errorf(codes.Internal, "Could not propose block: %v", err)
 	}
-
-	return bs.submitBlock(ctx, root, wrappedCapellaBlk)
+	return nil
 }
 
 func (bs *Server) submitBlock(ctx context.Context, blockRoot [fieldparams.RootLength]byte, block interfaces.ReadOnlySignedBeaconBlock) error {
```

Comment:
With this change I think SubmitBlockSSZ and SubmitBlindedBlockSSZ can be addressed and have submit block just removed and just use 1 such as propose beacon block. will be less logic all over the place. 
that being said I wonder if the unblind block logic is fine. at first glance it seems so and that seems to be the biggest difference between the submit block and proposeGenericBeaconBlock aside from the fact that the broadcaster is part is referenced from a different object.